File: src/interfaces/resumeCustomizer.js
'use strict';

// Strategy interface — swap the LLM behind resume tailoring.
class ResumeCustomizer {
  // eslint-disable-next-line no-unused-vars
  async customize(profile, jobTitle, company, jobDescription) {
    throw new Error('ResumeCustomizer subclasses must implement customize()');
  }
}

module.exports = { ResumeCustomizer };

File: python/src/interfaces/resume_customizer.py
from abc import ABC, abstractmethod
from ..models import UserProfile, CustomizedResume


class ResumeCustomizer(ABC):
    @abstractmethod
    async def customize(
        self,
        profile: UserProfile,
        job_title: str,
        company: str,
        job_description: str,
    ) -> CustomizedResume:
        pass
